src/app/pages/account/account.component.ts

```diff
@@ -15,34 +17,35 @@ export class AccountComponent implements OnInit {
   userIsAuthenticated: boolean = false;
   router: Router;
   subscription: Subscription;
+  orderList?: Map<number, Order[]> = new Map();
 
   constructor(
     private userService: UserService,
-    authService: AuthService,
+    private orderService: OrderService,
+    private authService: AuthService,
     router: Router
   ) {
     this.router = router;
-    this.subscription = authService.authenticationStatus.subscribe((status) => {
+
+  }
+
+  ngOnInit(): void {
+    this.subscription = this.authService.authenticationStatus.subscribe((status) => {
       this.userIsAuthenticated = status.valid;
       if (status.valid) {
-        userService.getUserDetails().subscribe((user) => {
+        /*userService.getUserDetails();*/
+        this.userService.getUserDetails().subscribe((user) => {
           this.user = user;
+          this.userService.updateUser(user);
           localStorage.setItem('userId', user.id.toString());
```

Comment:
This is added to local storage so that when creating an order we pass the id of the user that is placing the order and update the backend on whose specific order it is. We can do this with username as well if id is not preferable.